### ISSUE TITLE
[Bug]: Redundant index appconfig.ac_app_lazy_i

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/server#43103]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

```sql
create index ac_app_lazy_i
    on nextcloud.oc_appconfig (appid, lazy);

create index ac_app_lazy_key_i
    on nextcloud.oc_appconfig (appid, lazy, configkey);

```

the concatenated index ac_app_lazy_key_i fully covers the two first columns of ac_app_lazy_i. The database can also use ac_app_lazy_key_i when only the two first columns are accessed. ac_app_lazy_i is redundant. And hurts performance.

### Steps to reproduce

1. Install Nextcloud

### Expected behavior

Efficient use of database indices.

### Installation method

None

### Nextcloud Server version

master

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Additional info

Introduced with https://github.com/nextcloud/server/pull/41755.

:eyes: @ArtificialOwl and @artonge @come-nc @nickvergessen as reviewers

**Comment by nickvergessen** [comment]
WIP in https://github.com/nextcloud/server/pull/42903

**Comment by ArtificialOwl** [comment]
yes, should be fixed by #42903
